src/lib.rs

```diff
@@ -394,50 +400,30 @@ impl<T: tokio::io::AsyncSeek> futures_io::AsyncSeek for Compat<T> {
 }
 
 impl<T: futures_io::AsyncSeek> tokio::io::AsyncSeek for Compat<T> {
-    fn start_seek(
-        mut self: Pin<&mut Self>,
-        cx: &mut Context,
-        pos: io::SeekFrom,
-    ) -> Poll<io::Result<()>> {
-        let p = ready!(self.as_mut().project().inner.poll_seek(cx, pos))?;
-        *self.project().seek_res = Some(Ok(p));
-        Poll::Ready(Ok(()))
+    fn start_seek(mut self: Pin<&mut Self>, pos: io::SeekFrom) -> io::Result<()> {
+        *self.as_mut().project().seek_pos = Some(pos);
+        Ok(())
     }
 
-    fn poll_complete(self: Pin<&mut Self>, _: &mut Context) -> Poll<io::Result<u64>> {
-        Poll::Ready(
-            self.project()
-                .seek_res
-                .take()
-                .unwrap_or(Err(io::ErrorKind::Other.into())),
-        )
+    fn poll_complete(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<io::Result<u64>> {
+        let seek_pos = self
+            .seek_pos
+            .expect("poll_complete called without start_seek being called first");
```

Comment:
Hmm, this is not correct due to tokio 1.x AsyncSeek allows/recommends calling poll_complete before start_seek.
I'll send a fix.

Reply:
Filed #12